quadlet: fix runtime error for invalid Mount value

jakmeier opened the pull request:
If the `Mount` option inside a quadlet is missing the source=... part, the code today panics with the following message.

```
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/containers/podman/v5/pkg/systemd/quadlet.handleStorageSource(0xc000140de0?, 0x1d?, {0x0?, 0x1?}, 0x5634e39e233e?, 0x10?)
...
```

This commit checks for the missing source and returns an error to avoid the panic.

I added the check in two places, in handleStorageSource and in resolveContainerMountParams. The actual panic happens due to a unchecked index access in the former function while the porblematic configuration should already be detected before calling that function.

#### Does this PR introduce a user-facing change?

```release-note
Fix panic when a quadlet `Mount` option is missing a `source=...` value. Podman now reports a clear error instead of crashing.
```

baude commented:
thanks for your submission and congrats.  we appreciate any help.

the ci failed because we generally require a test for any PR submitted (or a modification of a test).  Is this something you think we can get a test for to protect against future regressions at the least?

Luap99 commented:
Note DCO check failed here because you use a different email for the sign off compared the the git author field

> Author: Jakob Meier, Committer: Jakob Meier; Expected "Jakob Meier [mail@jakobmeier.ch](mailto:mail@jakobmeier.ch)", but got "Jakob Meier [inbox@jakobmeier.ch](mailto:inbox@jakobmeier.ch)".

We can set it manually to pass but in general these values should be the same.

jakmeier commented:
Thanks everyone for the quick and helpful responses!

I've addresses all comments

- added a test case
- removed redundant check
- fixed sign off email

My new test case runs fine locally and CI seems to pass on that test, too.

I believe the remaining failing tests have nothing to do with my changes. Other PRs have the same failures.

Please take another look.

ygalblum reviewed (approved):
LGTM

Luap99 reviewed (approved):
/lgtm
